### PR TITLE
Refactor to a Superclass

### PR DESCRIPTION
**Pull request by joezuntz**

### CODE REVIEW

File: barber/base.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        # And convert that to a partition of zmin .. zmax
+        z_edges = np.concatenate([[self.zmin], self.zmin + self.zrange * np.cumsum(q)])
+        return z_edges
+
+    def evaluate_edges(self, bin_edges, extra_parameters):
+        if not self.quiet:
+            print(f"Edges: {bin_edges}   Extra: {extra_parameters}")
+
+        # Assign each galaxy in the training data to a bin
+        training_bin = np.digitize(self.training_z, bin_edges, right=True) - 1
```
Comment (aimalz) [inline barber/base.py:54]: I'm not sure I understand what `base.py` is doing.  Are we not defining "bin edges" solely in the space of photometric observations, without the training/test set distinction?  Or is this just for our own testing rather than any algorithm for grouping galaxies into bins?

**Review by aimalz — COMMENTED** [review]
Everything here makes sense except for some aspects of the superclass module itself that I don't understand.  However, I think its overall structure at this stage is sufficiently clear and unobjectionable for me to write up one of the unsupervised approach ideas as a subclass.  I suspect doing so will resolve my confusion.
